### PR TITLE
feat: support subpath module import

### PR DESCRIPTION
**Pull request by alexcodelf**
**Problem:**
Can not import subdirectory template from remote repository.

**Solution:**
Support subdirectory template import.
Example:
`https://github.com/alexcodelf/webservice//sub`
`https://github.com/alexcodelf/webservice//sub?ref=main`

The subdirectory is following double slashes.
**Related Issue:**
#template
#1339

### CODE REVIEW

File: pkg/vcs/repo.go [hunk 1]
```diff
@@ -58,6 +61,15 @@ func ParseURLToRepo(rawURL string) (*Repository, error) {
 		rawURL = strings.TrimSuffix(rawURL, "?ref="+ref)
 	}
 
+	// Get sub path from path.
+	if strings.Contains(path, "//") {
+		paths := strings.Split(path, "//")
+		subPath = paths[1]
+
+		path = strings.TrimSuffix(path, "//"+subPath)
+		rawURL = strings.TrimSuffix(rawURL, "//"+subPath)
```
Comment (gitlawr) [inline pkg/vcs/repo.go:70]: How do we handle sth like https://github.com/foo/bar//one//two?

Comment (alexcodelf) [reply pkg/vcs/repo.go:70]: In what cases will use this case? If it is a deep dir, it should be `https://github.com/alexcodelf/webservice//dep/sub`

Comment (gitlawr) [reply pkg/vcs/repo.go:70]: Not a valid use case. I ask because it will fail the parsing logic here. Do we have any validation to avoid this?

Comment (alexcodelf) [reply pkg/vcs/repo.go]: updated, create template will check git source parse.

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by aiwantaozi — DISMISSED** [review]

File: pkg/vcs/repo.go [hunk 2]
```diff
@@ -31,6 +31,8 @@ type Repository struct {
 	Topics []string `json:"topics"`
 	// Driver is the driver of the repository. E.G: github.
 	Driver string `json:"driver"`
+	// SubPath is the sub path of the repository.
+	SubPath string `json:"subPath"`
 }
 
 // ParseURLToGit parses a raw URL to a git repository.
```
Comment (aiwantaozi) [inline pkg/vcs/repo.go:38]: ParseURLToGit -> ParseURLToRepo

**Review by aiwantaozi — APPROVED** [review]

**Pull request merged**